controllers/numaresourcesoperator_controller.go

```diff
@@ -178,15 +179,19 @@ func (r *NUMAResourcesOperatorReconciler) reconcileResource(ctx context.Context,
 	var err error
 	err = r.syncNodeResourceTopologyAPI()
 	if err != nil {
+		r.Recorder.Eventf(instance, corev1.EventTypeWarning, "FailedAPISync", "failed to sync noderesourcetopology api: %v", err)
 		return ctrl.Result{}, status.ConditionDegraded, errors.Wrapf(err, "FailedAPISync")
 	}
+	r.Recorder.Eventf(instance, corev1.EventTypeNormal, "SuccessfulAPISync", "noderesourcetopology synchronization completed")
 
 	if r.Platform == platform.OpenShift {
 		// we need to create machine configs first and wait for the MachineConfigPool updates
 		// before creating additional components
 		if err := r.syncMachineConfigs(ctx, instance, mcps); err != nil {
+			r.Recorder.Eventf(instance, corev1.EventTypeWarning, "FailedMCSync", "failed to sync machine-config: %v", err)
 			return ctrl.Result{}, status.ConditionDegraded, errors.Wrapf(err, "failed to sync machine configs")
 		}
+		r.Recorder.Eventf(instance, corev1.EventTypeNormal, "SuccessfulMCSync", "machine-config synchronization completed")
```

Comment:
not sure about these two, inclined to yes though. The problem is again drawing a line between meaninful events and not spamming the users.